File: skills/rocketreach/scripts/check_account.js
#!/usr/bin/env node
/**
 * RocketReach Account Check
 * Shows account info, credit usage, and rate limits.
 *
 * Usage:
 *   export ROCKETREACH_API_KEY="your_key"
 *   node check_account.js
 */

const API_KEY = process.env.ROCKETREACH_API_KEY;
if (!API_KEY) {
  console.error("Error: set ROCKETREACH_API_KEY environment variable");
  process.exit(1);
}

async function main() {
  const res = await fetch("https://api.rocketreach.co/api/v2/account", {
    headers: { "Api-Key": API_KEY },
  });

  if (!res.ok) {
    console.error(`HTTP ${res.status}: ${await res.text()}`);
    process.exit(1);
  }

  const data = await res.json();

  console.log("ACCOUNT INFO");
  console.log("=".repeat(50));
  console.log(`  Name:   ${data.first_name || ""} ${data.last_name || ""}`);
  console.log(`  Email:  ${data.email || ""}`);
  console.log(`  Status: ${data.state || ""}`);

  console.log("\nCREDIT USAGE");
  console.log("-".repeat(50));
  for (const c of data.credit_usage || []) {
    console.log(
      `  ${c.credit_type.padEnd(25)} used: ${String(c.used).padEnd(8)} remaining: ${c.remaining}`
    );
  }

  console.log("\nRATE LIMITS");
  console.log("-".repeat(50));
  for (const r of data.rate_limits || []) {
    console.log(
      `  ${r.action.padEnd(20)} (${r.duration.padEnd(12)}) limit: ${String(r.limit).padEnd(6)} used: ${String(r.used).padEnd(4)} remaining: ${r.remaining}`
    );
  }
}

main().catch((e) => {
  console.error(`Error: ${e.message}`);
  process.exit(1);
});

File: skills/rocketreach/scripts/check_account.py
#!/usr/bin/env python3
"""
RocketReach Account Check
Shows account info, credit usage, and rate limits.

Usage:
  export ROCKETREACH_API_KEY="your_key"
  python3 check_account.py
"""

import json, urllib.request, ssl, os, sys

API_KEY = os.environ.get("ROCKETREACH_API_KEY")
if not API_KEY:
    print("Error: set ROCKETREACH_API_KEY environment variable")
    sys.exit(1)

url = "https://api.rocketreach.co/api/v2/account"
ctx = ssl.create_default_context()
req = urllib.request.Request(url)
req.add_header("Api-Key", API_KEY)
req.add_header("User-Agent", "curl/8.0")

try:
    resp = urllib.request.urlopen(req, context=ctx)
    data = json.loads(resp.read())

    print("ACCOUNT INFO")
    print("=" * 50)
    print(f"  Name:   {data.get('first_name','')} {data.get('last_name','')}")
    print(f"  Email:  {data.get('email','')}")
    print(f"  Status: {data.get('state','')}")

    print("\nCREDIT USAGE")
    print("-" * 50)
    for c in data.get("credit_usage", []):
        print(f"  {c['credit_type']:<25} used: {str(c['used']):<8} remaining: {c['remaining']}")

    print("\nRATE LIMITS")
    print("-" * 50)
    for r in data.get("rate_limits", []):
        print(f"  {r['action']:<20} ({r['duration']:<12}) limit: {str(r['limit']):<6} used: {str(r['used']):<4} remaining: {r['remaining']}")

except Exception as e:
    print(f"Error: {e}")
    sys.exit(1)
